internal/pkg/pluginmanager/manager.go

```diff
@@ -25,37 +27,49 @@ func DownloadPlugins(conf *configloader.Config) error {
 
 	for _, tool := range conf.Tools {
 		pluginFileName := configloader.GetPluginFileName(&tool)
+		pluginMD5FileName := configloader.GetPluginMD5FileName(&tool)
+		// plugin does not exist
 		if _, err := os.Stat(filepath.Join(pluginDir, pluginFileName)); errors.Is(err, os.ErrNotExist) {
-			// plugin does not exist
-			err := dc.download(pluginDir, pluginFileName, tool.Plugin.Version)
-			if err != nil {
+			// download .so file
```

Comment:
This part is duplicated code as the redownloadPlugins function.

Reply:
updated.
Here are the downloads separately, .so and .md5 whichever is missing, download whichever one is missing.
redownloadPlugins  function removes both .so and .md5 and download